cmd/soroban-rpc/internal/methods/get_version_info.go

```diff
@@ -0,0 +1,42 @@
+package methods
+
+import (
+	"context"
+	"github.com/creachadair/jrpc2"
+	"github.com/creachadair/jrpc2/handler"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/config"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/daemon/interfaces"
+)
+
+type GetVersionRequest struct {
+}
+
+type GetVersionResponse struct {
+	Version                string `json:"version"`
+	CommitHash             string `json:"commit_hash"`
+	BuildTimestamp         string `json:"build_time_stamp"`
+	ProtocolVersion        int    `json:"protocol_version"`
+	XDRVersion             string `json:"xdr_version"`
+	CaptiveCoreVersionInfo string `json:"captive_core_version"`
+}
+
+func NewGetVersionInfoHandler(daemon interfaces.Daemon) jrpc2.Handler {
+	coreClient := daemon.CoreClient()
+	return handler.New(func(ctx context.Context, request GetVersionRequest) (GetVersionResponse, error) {
+		info, err := coreClient.Info(ctx)
```

Comment:
maybe it's better that we still return as much as we can in this case? so then, for example, if the Captive Core is bogged down, we can still provide version info

Reply:
actually, giving user the ability to talk to Core at all is dangerous - it's not built to handle requests like that. since we're just using it for the protocol version, we should cache that on startup, instead!

Reply:
Both great points !!

Reply:
Removed ProtocolVersion from response: From stellar-rpc document, it seems like we also want `passphrase` along with `protocol version`, so it might be just better to use `getNetwork` for fetching protocol version. (We could merge both the endpoints later in future but keeping it out of scope for this PR)